\documentclass[sigconf]{acmart}
\usepackage{booktabs} % For formal tables
\usepackage{wrapfig, framed}

% Copyright
%\setcopyright{none}
%\setcopyright{acmcopyright}
%\setcopyright{acmlicensed}
\setcopyright{rightsretained}
%\setcopyright{usgov}
%\setcopyright{usgovmixed}
%\setcopyright{cagov}
%\setcopyright{cagovmixed}


% DOI
\acmDOI{xx.xxx/xxxx}

% ISBN
\acmISBN{xxxx-xxxx-xxxx-xxxx}

%Conference
\acmConference[XXXX'18]{ACM conference}{xxx 2018}{xxxx, xx USA}
\acmYear{2018}
\copyrightyear{2018}


\acmArticle{xx}
\acmPrice{xx}

% These commands are optional
%\acmBooktitle{Transactions of the ACM Woodstock conference}
\editor{xxxxx}
\editor{xxxxx}


\begin{document}
\title{Impact of Pre-processing in Predicting Neuron Cell Clusters from Single-Cell RNA-Seq data}

\author{Sajal Kumar}
\affiliation{%
  \institution{Department of Computer Science,\\New Mexico State University}
  \streetaddress{1780 E University Ave}
  \city{Las Cruces}
  \state{New Mexico}
  \postcode{88003}
}
\email{sajal49@nmsu.edu}

\author{Jiandong Wang}
\affiliation{%
  \institution{Department of Computer Science,\\New Mexico State University}
  \streetaddress{1780 E University Ave}
  \city{Las Cruces}
  \state{New Mexico}
  \postcode{88003}
}
\email{wangjd24@nmsu.edu}

\author{Xiaonan Zhu}
\affiliation{%
  \institution{Department of Mathematical Sciences,\\New Mexico State University}
  \streetaddress{1780 E University Ave}
  \city{Las Cruces}
  \state{New Mexico}
  \postcode{88003}
}
\email{xzhu@nmsu.edu}
% The default list of authors is too long for headers.
\renewcommand{\shortauthors}{S. Kumar et al.}


\begin{abstract}
Single-cell RNA sequencing (scRNA-seq) technology is un-parallel in providing high resolution gene expression data at cellular levels. However, it is also notoriously known for being noisy due to amplification and dropout, making it a challenge for analytical methods to differentiate noise from patterns. Here we employ a simple decision tree algorithm to evaluate the difficulty of using Single Cell RNA-Seq data to predict mice neuron cell clusters by gene expression and explore the effect of various pre-processing strategies on the predictive accuracy. We use gene profiles of 20-K neuron single cells measured from two mice, during embryonic day 18, that capture cells from three different regions of the brain. We begin by applying some simple and standard pre-processing strategies on the data set. Having no information on the spatial location of the cells, we next use AGNES hierarchical clustering to group the single cells into three clusters. With the deduced cell regions acting as class-labels, we perform the task of classification based on gene expression dynamics and report the 10-fold accuracy, we consider this the baseline. We then transform the data using two scalable imputation and denoising techniques : 1) DCA -- a deep neural network based denoising auto-encoder and 2) MAGIC -- Markov Affinity-based Graph Imputation of Cells, followed by respective runs of the classification task. Based on empirical evidence, the DCA transformation can better capture the inherent dynamics of the data-set and leads to significant improvement in accuracy from the baseline, however, MAGIC over-fits the data, resulting in sub par performance when compared to the baseline. Our analysis provides an unbiased third party validation of the standard practices in Single Cell and the effectiveness of the DCA algorithm, while also revealing the demerits of MAGIC.
\end{abstract}

%
% The code below should be generated by the tool at
% http://dl.acm.org/ccs.cfm
% Please copy and paste the code instead of the example below.
%
\begin{CCSXML}
<ccs2018>
 <concept>
  <concept_id>10010520.10010553.10010562</concept_id>
  <concept_desc>Computational biology~Single cell RNA-Seq</concept_desc>
  <concept_significance>300</concept_significance>
 </concept>
 <concept>
  <concept_id>10010520.10010575.10010755</concept_id>
  <concept_desc>Computational biology~Cell Classification</concept_desc>
  <concept_significance>500</concept_significance>
 </concept>
 <concept>
  <concept_id>10010520.10010553.10010554</concept_id>
  <concept_desc>Computational biology~Denoising</concept_desc>
  <concept_significance>500</concept_significance>
 </concept>
 <concept>
  <concept_id>10003033.10003083.10003095</concept_id>
  <concept_desc>Computational biology~Deep autoencoders</concept_desc>
  <concept_significance>300</concept_significance>
 </concept>
</ccs2018>
\end{CCSXML}

\ccsdesc[500]{Computational biology~Cell Classification}
\ccsdesc[500]{Computational biology~Denoising}
\ccsdesc[300]{Computational biology~Single cell RNA-Seq}
\ccsdesc[300]{Computational biology~Deep autoencoders}
\ccsdesc[300]{Computational biology~Markov Affinity-based Graph}

\keywords{Cell Classification, Denoising, Single cell RNA-Seq, Deep autoencoders, Markov Affinity-based Graph}

\maketitle

\input{samplebody-conf}

\bibliographystyle{ACM-Reference-Format}
\bibliography{sample-bibliography}

\end{document}