\documentclass[sigconf]{acmart}

\usepackage{booktabs} % For formal tables


% Copyright
%\setcopyright{none}
%\setcopyright{acmcopyright}
%\setcopyright{acmlicensed}
\setcopyright{rightsretained}
%\setcopyright{usgov}
%\setcopyright{usgovmixed}
%\setcopyright{cagov}
%\setcopyright{cagovmixed}


% DOI
\acmDOI{xx.xxx/xxxx}

% ISBN
\acmISBN{xxxx-xxxx-xxxx-xxxx}

%Conference
\acmConference[XXXX'18]{ACM conference}{xxx 2018}{xxxx, xx USA}
\acmYear{2018}
\copyrightyear{2018}


\acmArticle{xx}
\acmPrice{xx}

% These commands are optional
%\acmBooktitle{Transactions of the ACM Woodstock conference}
\editor{xxxxx}
\editor{xxxxx}


\begin{document}
\title{Using deep autoencoders for denoising and reduction of Single-Cell RNA-Seq data}

\author{Sajal Kumar}
\affiliation{%
  \institution{Department of Computer Science,\\New Mexico State University}
  \streetaddress{1780 E University Ave}
  \city{Las Cruces}
  \state{New Mexico}
  \postcode{88003}
}
\email{sajal49@nmsu.edu}

\author{Jiandong Wang}
\affiliation{%
  \institution{Department of Computer Science,\\New Mexico State University}
  \streetaddress{1780 E University Ave}
  \city{Las Cruces}
  \state{New Mexico}
  \postcode{88003}
}
\email{wangjd24@nmsu.edu}

\author{Xiaonan Zhu}
\affiliation{%
  \institution{Department of Mathematical Sciences,\\New Mexico State University}
  \streetaddress{1780 E University Ave}
  \city{Las Cruces}
  \state{New Mexico}
  \postcode{88003}
}
\email{xzhu@nmsu.edu}
% The default list of authors is too long for headers.
\renewcommand{\shortauthors}{S. Kumar et al.}


\begin{abstract}
Single-cell RNA sequencing (scRNA-seq) technology is un-parallel in providing high resolution gene expression data at cellular levels. However, it is also notoriously known for being noisy due to amplification and dropout, making it a challenge for analytical methods to differentiate noise from patterns, so a scalable imputation and denoising mechanism is required. Here we apply DCA -- a deep neural network based denoising auto-encoders for scRNA-seq on a brain cell dataset from 10x genomics, which profiles 1.3 M cells from embryonic mice brains. We use this data set to see if gene expression can be used to classify various brain-cell types and illustrate the performance of the classical decision tree method before and after encoding.
\end{abstract}

%
% The code below should be generated by the tool at
% http://dl.acm.org/ccs.cfm
% Please copy and paste the code instead of the example below.
%
\begin{CCSXML}
<ccs2018>
 <concept>
  <concept_id>10010520.10010553.10010562</concept_id>
  <concept_desc>Computational biology~Single cell RNA-Seq</concept_desc>
  <concept_significance>300</concept_significance>
 </concept>
 <concept>
  <concept_id>10010520.10010575.10010755</concept_id>
  <concept_desc>Computational biology~Dimensionality reduction</concept_desc>
  <concept_significance>500</concept_significance>
 </concept>
 <concept>
  <concept_id>10010520.10010553.10010554</concept_id>
  <concept_desc>Computational biology~Denoising</concept_desc>
  <concept_significance>500</concept_significance>
 </concept>
 <concept>
  <concept_id>10003033.10003083.10003095</concept_id>
  <concept_desc>Computational biology~Deep autoencoders</concept_desc>
  <concept_significance>300</concept_significance>
 </concept>
</ccs2018>
\end{CCSXML}

\ccsdesc[500]{Computational biology~Dimensionality reduction}
\ccsdesc[500]{Computational biology~Denoising}
\ccsdesc[300]{Computational biology~Single cell RNA-Seq}
\ccsdesc[300]{Computational biology~Deep autoencoders}

\keywords{Dimensionality reduction, Denoising, Single cell RNA-Seq, Deep autoencoders}

\maketitle

\input{samplebody-conf}

\bibliographystyle{ACM-Reference-Format}
\bibliography{sample-bibliography}

\end{document}