\section*{Introduction}

\textbf{Motivation:} Single-cell RNA-seq is a promising technique that profiles transcriptomes over a plethora of cell types. By measuring individual cells, rather than averaged tissue types, scRNA-seq can uncover biological mechanisms that is not observed by the average behaviors of a bulk of cells  \citep{wang2017vasc}. However, despite improvements in measuring technologies, factors including amplification bias, cell cycle effects \citep{buettner2015computational}, etc. lead to substantial noise in scRNA-seq experiments. The low RNA capture rate leads to failure of detection of an expressed gene resulting in a ``false" zero count observation, defined as dropout event; which potentially corrupts the underlying biological signal \cite{eraslan2018single}. One such scRNA-seq data set is from the 10x genomics project that  profiles 1.3 M embroynic brain cells of mice. It represents cells from cortex, hippocampus and subventricular zone of two mice at 18 days post conception which is an invaluable resource to study gene dynamics during an important development stage, however, the corruption induced by dropout events compromises our ability to use analytical techniques to extract meaningful patterns. 

Current approaches for scRNA-seq denoising / imputation rely on using the correlation structure of single-cell gene expression data, leveraging similarities between cells and/or genes that does not scale well; or has explicit / implicit linearity assumption that may miss complex non-linear patterns \citep{eraslan2018single}. Deep-learning auto-encoders can learn the underlying complex manifold, that represents the biological processes and/or cellular states and utilize that to reduce the high dimensional data space to significantly lower dimensions \citep{moon2017manifold}. However, using an auto-encoder as is may fail due to the noise model not being adapted to typical scRNA-seq noise. One such autoencoder DCA -- deep count autoencoder facilitates non-linear embedding of cells and uses scRNA-seq data specific loss function based on negative binomial count distributions which offers adaptive learning and encoding of the data manifold \citep{eraslan2018single}. Thus, we utilize DCA to denoise and reduce the dataset and then apply the classical decision tree algorithms to learn if gene expression dynamics can infer different cell types.

\textbf{Problem definition:} Given expression of X genes across m cell types, we denoise and reduce the data set to X' gene expression of m cell types, expecting $n(X') \leq n(X)$ by using the DCA autoencoder. We then apply the decision tree classifier on X' (genes expressions) to learn the m classes (cell types). We also apply the decision tree algorithm on the raw data to see if the encoded X' actually results in improved accuracy.

\section*{Solution}

\subsection*{Data source}
We are using Single Cell Gene Expression Dataset by Cell Ranger 1.3.0 from the 10x Genomics project that provides 1.3 Million Brain Cells from E18 Mice which consists of cells from cortex, hippocampus and subventricular zone of two E18 mice. The data was retrieved from \href{https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.3.0/1M_neurons}{\textcolor{blue}{support.10xgenomics.com}}.

\begin{figure*}
\begin{framed}
\begin{minipage}{\linewidth}
\begin{minipage}[t]{0.48\linewidth}
\vspace{0pt}
\textsf{a)}\\
\includegraphics[width=\linewidth]{without_autoencoding.pdf}
\end{minipage}
\begin{minipage}[t]{0.48\linewidth}
\vspace{0pt}
\textsf{b)}\\
\includegraphics[width=\linewidth]{with_autoencoding.pdf}
\end{minipage}
\end{minipage}
\caption{a) In the first analysis without any denoising, the single cell data will be supplied as is to a sklearn decision tree algorithm. b) In the second analysis, the single cell data will be denoised using the ZINB DCA autoencoder and then the learned manifold X$^{'}$ will be supplied to the sklearn decision tree algorithm.}
\end{framed}
\label{fig1}
\end{figure*}

\subsection*{Proposed Pipeline}
We will proceed with two phases of analysis as shown in figure~1. The first phase would include using the single cell data 'as-is' ($X$, $Y$) and learn if there exists a non-linear combination of gene expression $M(X)$ that can differentiate between the three class labels ($Y$) : \textit{cortex}, \textit{hippo-campus} and  \textit{sub-ventricular zone}. We will be utilizing the decision tree classifier -- \textit{sklearn.tree.DecisionTreeClassifier} provided by sci-kit learn package in Python. We will use sklearn.model\_selection to perform the 10-fold cross-validation strategy and evaluate the performance of the decision tree model.

Next, we will use the zero-inflated negative binomial (ZINB) DCA auto-encoder \citep{eraslan2018single} to first denoise the single cell data ($X$) and see if we can learn the manifold of the data, as claimed by the authors of this method. The augmented data $X'$ will then be given to a similar setup of decision-tree learning with 10-fold cross validation to learn the three cell types. Our final objective would be to report any improvement in performance, followed by deeper insight on to why/why didn't the auto-encoding worked. Our expectation is that the augmented data $X'$ should work better as it uses zero-inflated negative binomial distribution that can model highly sparse and over-dispersed data \citep{eraslan2018single} which makes it very suitable for single-cell data.
